cocotb/share/lib/embed/gpi_embed.cpp

```diff
@@ -131,7 +131,6 @@ extern "C" void embed_init_python(void)
     set_program_name_in_venv();
     Py_Initialize();                    /* Initialize the interpreter */
     PySys_SetArgvEx(1, argv, 0);
-    PyEval_InitThreads();               /* Create (and acquire) the interpreter lock */
```

Comment:
Even though this is unnecessary since Python 3.7+, we still need it for 3.5 and 3.6, if I'm reading the docs at https://docs.python.org/3/c-api/init.html#c.PyEval_InitThreads correctly. Can you gate it with an ifdef?

Reply:
Or we could just leave it in:

> Changed in version 3.9: The function now does nothing.

Reply:
I'd re-read the second sentence in the documentation carefully. And we cannot leave it in, it is marked as deprecated and `-Werror` causes the CI run to fail.

Reply:
Ah yes. What a confusing piece of documentation.